app/models/spree/order_inventory_assembly.rb

```diff
@@ -21,8 +21,9 @@ def verify(shipment = nil)
           self.variant = part
 
           if existing_parts < total_parts
-            shipment ||= determine_target_shipment
-            add_to_shipment(shipment, total_parts - existing_parts)
+            quantity = total_parts - existing_parts
+            shipment ||= determine_target_shipment(quantity)
```

Comment:
I think that to support every Solidus version we should handle both cases with something like:
```ruby

      def determine_target_shipment(quantity)
        if determine_target_shipment_requires_quantity?
          super(quantity)
        else
          super()
        end
      end

      private

      def determine_target_shipment_requires_quantity?
         OrderInventory.instance_method(:determine_target_shipment).arity != 0
      end
    end
```